src/sqlfluff/rules/references/RF05.py

```diff
@@ -101,6 +104,15 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         ):
             return LintResult(memory=context.memory)
 
+        if (
+            context.parent_stack
+            and context.parent_stack[-1].is_type("column_reference")
+            and context.parent_stack[-2].is_type("select_clause_element")
+            and context.parent_stack[-2].get_child("alias_expression")
+        ):
+            # If selects are aliased, ignore unaliased column reference
+            return None
```

Comment:
Is your logic here that if a column is aliased, then we likely don't have control over the _reference_, only the _alias_, and so we should only lint the content of the alias?

I hadn't thought about it like that, but I think I like it 🤔 .

If that is your intent however, I think we need a more robust algorithm here, in particular, I think this would break if the reference was in a function (like `CAST("'PIVOTED1'" AS INT)`). Could I suggest instead, iterating up the `parent_stack` until a `select_clause_element` is found (and then check for an `alias_expression` relative to it, in the way you've already done)? In doing that you might want to add a condition to stop iterating of either a) running out of parents _or_ if you encounter a `statement` segment then also stop because we're going too far?

How does that sound?

Reply:
https://github.com/sqlfluff/sqlfluff/pull/6297/commits/18c2b45e605576ace1cf6cfe965bc65bde339c9c :)